geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSUnionIntegrationTest.java

```diff
@@ -56,31 +63,122 @@ public void sunionErrors_givenTooFewArguments() {
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String[] result = {"one", "two", "three", "four", "five", "linux", "peach"};
+    assertThat(jedis.sunion(setKey1, setKey2)).containsExactlyInAnyOrder(result);
+  }
+
+  @Test
+  public void sunionWithSetsWithAllSharedValues_returnsUnionOfSets() {
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, setMembers1);
+
+    assertThat(jedis.sunion(setKey1, setKey2)).containsExactlyInAnyOrder(setMembers1);
+  }
+
+  @Test
+  public void sunionWithMultipleSets_returnsUnionOfSets() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    String thirdSetKey = "{user1}setKey3";
+    String[] thirdSetMembers = new String[] {"fire", "dance", "floor", "five"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+    jedis.sadd(thirdSetKey, thirdSetMembers);
+
+
+    String[] result =
+        {"one", "two", "three", "four", "five", "linux", "peach", "fire", "dance", "floor"};
+    assertThat(jedis.sunion(setKey1, setKey2, thirdSetKey))
+        .containsExactlyInAnyOrder(result);
+  }
+
+  @Test
+  public void sunionWithSetsFromDifferentSlots_returnsCrossSlotError() {
+    String setKeyDifferentSlot = "{user2}setKey2";
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKeyDifferentSlot, secondSetMembers);
+
+    assertThatThrownBy(() -> jedis.sunion(setKey1, setKeyDifferentSlot))
+        .hasMessageContaining(ERROR_DIFFERENT_SLOTS);
+  }
+
+
+  @Test
+  public void sunionWithDifferentKeyType_returnsWrongTypeError() {
+    jedis.set("ding", "dong");
+    assertThatThrownBy(() -> jedis.sunion("ding")).hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void ensureSetConsistency_whenRunningConcurrently() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String[] unionMembers = {"one", "two", "three", "four", "five", "linux", "peach"};
+    Set<String> unionList = new HashSet<>(Arrays.asList(unionMembers));
+    Set<String> firstSetMembersList = new HashSet<>(Arrays.asList(setMembers1));
+
+
+    final AtomicReference<Set<String>> sunionResultReference = new AtomicReference<>();
+    new ConcurrentLoopingThreads(1000,
+        i -> jedis.srem(setKey2, secondSetMembers),
+        i -> sunionResultReference.set(jedis.sunion(setKey1, setKey2)))
+            .runWithAction(() -> {
+              assertThat(sunionResultReference).satisfiesAnyOf(
+                  sunionResult -> assertThat(sunionResult.get())
+                      .containsExactlyInAnyOrderElementsOf(firstSetMembersList),
+                  sunionResult -> assertThat(sunionResult.get())
+                      .containsExactlyInAnyOrderElementsOf(unionList));
```

Comment:
These assertions can be simplified to:
```
              assertThat(sunionResultReference).satisfiesAnyOf(
                  sunionResult -> assertThat(sunionResult.get())
                      .containsExactlyInAnyOrder(setMembers1),
                  sunionResult -> assertThat(sunionResult.get())
                      .containsExactlyInAnyOrder(unionMembers));
```
which means we no longer need the `unionList` and `firstSetMembersList` variables.

Reply:
I couldn't put those directly because it was it was saying I needed an iterator.

Reply:
When I use the code pasted above, the test works fine. Note that it's using `containsExactlyInAnyOrder()` rather than `containsExactlyInAnyOrderElementsOf()`.